chapters/packages.tex

```diff
@@ -321,3 +321,132 @@ \section{External Resources}\label{external-resources}
 \filename{modelica://Modelica/Mechanics/C.jpg} is illegal.
 The references \filename{modelica://Modelica.Mechanics/C.jpg} and \filename{modelica://Modelica/C.jpg} must also refer to two distinct resources.
 \end{example}
+
+\section{Multilingual Descriptions}\label{multilingual-descriptions}
+
+\begin{nonnormative}
+Descriptive texts in a model or library are usually formulated in English.
+To support users who are not so familiar with this, a Modelica tool can support the translation of these texts into any language.
+This should be done using external files, so that no modification of the actual Modelica text is required.
+\end{nonnormative}
+
+The texts in following Modelica constructs should be translated:
+\begin{itemize}
+\item description strings of component declarations and classes
+\item strings in the following annotations:
+\begin{itemize}
+\item \lstinline!Text.string!, \lstinline!Text.textString!
+\item \lstinline!missingInnerMessage!, \lstinline!obsolete!, \lstinline!unassignedMessage!
+\item \lstinline!Dialog.group!, \lstinline!Dialog.tab!
+\item \lstinline!Dialog.loadSelector.caption!, \lstinline!Dialog.loadSelector.filter!, \lstinline!Dialog.saveSelector.caption!, \lstinline!Dialog.saveSelector.filter!
+\item \lstinline!Documentation.info!, \lstinline!Documentation.revisions!
+\item \lstinline!Figure.title!, \lstinline!Plot.title!, \lstinline!Curve.legend!
+\end{itemize}
+\end{itemize}
+
+C-style and line comments are not translated.
+
+The files to support translation must be provided by the developer of the library.
+They must be stored in the resources directory \filename{modelica://$\mathit{LibraryName}$/Resources/Language/}.
+
+Two kind of files have to be provided:
+\begin{enumerate}
+\item Template file \filename{$\mathit{LibraryName}$.pot}, as the resource \filename{modelica://$\mathit{LibraryName}$/Resources/Language/$\mathit{LibraryName}$.pot}.
+It contains all necessary information to translate all descriptions, but no translations.
+The pattern \filename{$\mathit{LibraryName}$} denotes the toplevel class name of the library.
+\item One file for each supported language with the name \filename{$\mathit{LibraryName}$.$\mathit{language}$.po}, as the resource \filename{modelica://$\mathit{LibraryName}$/Resources/Language/$\mathit{LibraryName}$.$\mathit{language}$.po}.
+This file is a copy of the associated template file, but extended with the translations in the specified language.
+The pattern \filename{$\mathit{language}$} stands for the ISO 639-1 language code, e.g., \filename{de} or \filename{en}.
+\end{enumerate}
+
+The detailed format of these files is described in \href{https://www.gnu.org/software/gettext/manual/gettext.pdf}{GNU gettext}.
+For Modelica translations, only the keywords \lstinline!msgctxt!, \lstinline!msgid! and \lstinline!msgstr! are used, meaning that a translation entry looks like this:
+\begin{lstlisting}
+#: $\mathit{filename}$:$\mathit{lineNumber}$
```

Comment:
To avoid incorrect treatment in translation tools, we should make sure we don't get C interpretation:
```suggestion
#: $\mathit{filename}$:$\mathit{lineNumber}$
# no-c-format
```